core/app/components/FormBuilder/FormBuilder.tsx

```diff
@@ -107,13 +114,98 @@ type Props = {
 	stages: Stages[];
 };
 
+/**
+ * Only sends the dirty fields to the server
+ */
+const preparePayload = ({
+	formValues,
+	defaultValues,
+}: {
+	defaultValues: FormBuilderSchema;
+	formValues: FormBuilderSchema;
+}) => {
+	const { upserts, deletes, relatedPubTypes, deletedRelatedPubTypes } =
+		formValues.elements.reduce<{
+			upserts: NewFormElements[];
+			deletes: FormElementsId[];
+			relatedPubTypes: NewFormElementToPubType[];
+			deletedRelatedPubTypes: FormElementsId[];
+		}>(
+			(acc, element, index) => {
+				if (element.deleted) {
+					if (element.elementId) {
+						acc.deletes.push(element.elementId);
+					}
+				} else if (!element.elementId) {
+					// Newly created elements have no elementId, so generate an id to use
+					const id = crypto.randomUUID() as FormElementsId;
+					acc.upserts.push(
+						formElementsInitializerSchema.parse({
+							formId: formValues.formId,
+							...element,
+							id,
+						})
+					);
+					if (element.relatedPubTypes) {
+						for (const pubTypeId of element.relatedPubTypes) {
+							acc.relatedPubTypes.push({ A: id, B: pubTypeId });
+						}
+					}
+				} else if (element.updated) {
+					// check whether the element is reeeaally updated minus the updated field
+					const { updated: _, id: _id, ...elementWithoutUpdated } = element;
+					const { updated, id, ...rest } =
+						defaultValues.elements.find((e) => e.elementId === element.elementId) ?? {};
+
+					const defaultElement = rest as Omit<FormElementData, "updated" | "id">;
+
+					if (JSON.stringify(defaultElement) === JSON.stringify(elementWithoutUpdated)) {
+						return acc;
+					}
+
+					acc.upserts.push(
+						formElementsInitializerSchema.parse({
+							...element,
+							formId: formValues.formId,
+							id: element.elementId,
+						})
+					); // TODO: only update changed columns
+					if (element.relatedPubTypes) {
+						// If we are updating to an empty array and there were related pub types before, we should clear out all related pub types
+						if (
+							element.relatedPubTypes.length === 0 &&
+							defaultElement.relatedPubTypes?.length
+						) {
```

Comment:
i added this `defaultElement.relatedPubTypes?.length` check, bc otherwise if you swapped a relation element back and forth it would assume something had to be deleted, which didn't make sense